### DISCUSSION TITLE
How can I enable numbering for my figures for a reveal js presentation?

### DISCUSSION THREAD

**Discussion by Martin15135215** [quarto-dev/quarto-cli#7729] [Q&A] [answered]
### Description

I look at here in the documentation:
- https://quarto.org/docs/reference/formats/presentations/revealjs.html#numbering
- https://quarto.org/docs/reference/formats/presentations/revealjs.html#figures

I only saw the option to add numbering for the section.

**Comment by mcanouil** [comment, marked as answer]
You need to make your figures as cross-reference.

**Reply by Martin15135215** [reply]
Thanks! I did try it with a PNG file and for a mermaid code block. Why is the caption from a mermaid diagram way smaller than of caption of the normal figure?

**Reply by mcanouil** [reply]
You'll have to provide a small reproducible example.

**Reply by Martin15135215** [reply]
![image](https://github.com/quarto-dev/quarto-cli/assets/110968398/6edeae7d-9778-418c-9105-eb330df0f30d)
![image](https://github.com/quarto-dev/quarto-cli/assets/110968398/8d08e1ea-cb0e-46d3-8126-edab7a3b3390)

I am not sure what is the best method to share a ipynb file, so I did a screenshot and here are they pasted, to facilitate easy copy/pasting:


in raw cell:
```raw
---
title: Test
author: Martin
date: 2023-12-01
format:
  revealjs: 
    theme: dark
    slide-number: true
    mermaid:
      theme: dark
    standalone: true
    execute:
      echo: true
---
```

in markdown cell:
```markdown
## Mermaid
```

in markdown cell:
```{mermaid}
%%| echo: false
%%| label: fig-mermaid
%%| fig-cap: "Mermaid"
flowchart TD
    A[Christmas] -->|Get money| B(Go shopping)
    B --> C{Let me think}
    C -->|One| D[Laptop]
    C -->|Two| E[iPhone]
    C -->|Three| F[fa:fa-car Car]
```


in markdown cell:
```markdown
## Picutre
```

in markdown cell:
```markdown
![lorenzo-hamers-oPqdoLy0MDo-unsplash.jpg](attachment:0b822cde-8199-4f02-ad8d-a464bd555a30.jpg){#fig-picutre}
```

The photo is from Unsplash:
![lorenzo-hamers-oPqdoLy0MDo-unsplash](https://github.com/quarto-dev/quarto-cli/assets/110968398/42ef173b-710f-4b3b-8100-39e2439db332)

That the presentation:

![image](https://github.com/quarto-dev/quarto-cli/assets/110968398/d408cc4d-3425-41ca-8461-26cf5ca4267b)

![image](https://github.com/quarto-dev/quarto-cli/assets/110968398/21af4dc3-d243-49f2-8eb8-88e30847dc8a)


The caption have different caption size

**Reply by mcanouil** [reply]
You can share a Quarto document using the following syntax, _i.e._, using more backticks than you have in your document (usually four ` ```` `).

`````md
````qmd
---
title: "Reproducible Quarto Document"
format: html
---

This is a reproducible Quarto document using `format: html`.
It is written in Markdown and contains embedded R code.
When you run the code, it will produce a plot.

```{r}
plot(cars)
```

![A placeholder image](https://placehold.co/600x400.png)

The end.
````
`````

Use your favourite browser inspect mode and use CSS to modify components.
Also look at the documentation guide for revealjs which has everything regarding advanced theming.

Note that the difference of size is likely due to reactivity / auto scale/resize which is also described in the documentation.

See:
- https://quarto.org/docs/authoring/diagrams.html#sizing
- https://quarto.org/docs/presentations/revealjs/advanced.html#stretch

**Reply by Martin15135215** [reply]
I used `quarto convert Test.ipynb`

`````md
````qmd
---
title: Test
author: Martin
date: '2023-12-01'
format:
  revealjs:
    theme: dark
    slide-number: true
    mermaid:
      theme: dark
    standalone: true
    execute:
      echo: true
jupyter: python3
---

## Mermaid

```{mermaid}
%%| echo: false
%%| label: fig-mermaid
%%| fig-cap: "Mermaid"
flowchart TD
    A[Christmas] -->|Get money| B(Go shopping)
    B --> C{Let me think}
    C -->|One| D[Laptop]
    C -->|Two| E[iPhone]
    C -->|Three| F[fa:fa-car Car]
```

## Picutre

![lorenzo-hamers-oPqdoLy0MDo-unsplash.jpg](attachment:0b822cde-8199-4f02-ad8d-a464bd555a30.jpg){#fig-picutre}
````
`````

**Reply by mcanouil** [reply]
no need to have ` ````qmd `, it was for the example or using more backticks.

**Reply by Martin15135215** [reply]
> Use your favourite browser inspect mode and use CSS to modify components.
> Also look at the documentation guide for revealjs which has everything regarding advanced theming.
> 
> Note that the difference of size is likely due to reactivity / auto scale/resize which is also described in the documentation.

I'm not familiar with HTML/CSS/Javasript...

**Reply by mcanouil** [reply]
Unfortunately, at some point if you don't want the default of HTML based format, you'll have to learn.
There never will be ways around that.

This being said, here it seems there is something fishy.
Could you open a new discussion with the example (but without the surrounding ` ````qmd `)?

**Reply by Martin15135215** [reply]
I am pleased with the default dark theme. I found it just confusing why the fig caption size is inconsistent. Therefore, I thought this was not intended, more a bug.

> Could you open a new discussion with the example (but without the surrounding ````qmd)?

See: https://github.com/quarto-dev/quarto-cli/discussions/7763
